src/shell/commands/executable.rs

```diff
@@ -55,22 +57,33 @@ impl ShellCommand for ExecutableCommand {
       // avoid deadlock since this is holding onto the pipes
       drop(sub_command);
 
-      tokio::select! {
-        result = child.wait() => match result {
-          Ok(status) => ExecuteResult::Continue(
-            status.code().unwrap_or(1),
-            Vec::new(),
-            Vec::new(),
-          ),
-          Err(err) => {
-            let _ = stderr.write_line(&format!("{}", err));
-            ExecuteResult::Continue(1, Vec::new(), Vec::new())
-          }
-        },
-        _ = context.state.token().cancelled() => {
+      let child_clone = Arc::clone(&child);
+
+      tokio::spawn(async move {
+        tokio::signal::ctrl_c().await.unwrap();
+        let mut child = child_clone.lock().await;
+        if let Ok(None) = child.try_wait() {
           let _ = child.kill().await;
-          ExecuteResult::for_cancellation()
         }
+      });
+
+      let mut child_locked = child.lock().await;
+      tokio::select! {
+          result = child_locked.wait() => match result {
+              Ok(status) => ExecuteResult::Continue(
+                  status.code().unwrap_or(1),
+                  Vec::new(),
+                  Vec::new(),
+              ),
+              Err(err) => {
+                  let _ = stderr.write_line(&format!("{}", err));
+                  ExecuteResult::Continue(1, Vec::new(), Vec::new())
+              }
+          },
+          _ = context.state.token().cancelled() => {
+              let _ = child_locked.kill().await;
+              ExecuteResult::for_cancellation()
```

Comment:
I kind of wonder if we should just add a single listener in Deno for ctrl+c then cancel the token on `ShellState` (we'd have to manually set it up to use https://docs.rs/deno_task_shell/0.18.1/deno_task_shell/fn.execute_with_pipes.html though).